src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -512,6 +513,174 @@ pass::EliminateConcat::EliminateConcat() {
     this->register_matcher(m, callback);
 }
 
+pass::EliminateConcatSplit::EliminateConcatSplit() {
+    MATCHER_SCOPE(EliminateConcatSplit);
+    auto pattern_concat = pattern::wrap_type<ov::op::v0::Concat>();
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_map();
+        const auto concat = ov::as_type_ptr<ov::op::v0::Concat>(pattern_map.at(pattern_concat));
+        if (!concat || concat->is_dynamic() || concat->get_users().size() == 1) {
```

Comment:
`concat->is_dynamic() || concat->get_users().size() == 1` it's better to move this condition to the predicate.

auto pattern_concat = pattern::wrap_type<ov::op::v0::Concat>(**predicate**);